baserun/instrumentation/base_instrumentor.py

```diff
@@ -57,7 +57,11 @@ def _instrument(self, **kwargs):
                 f"{original_class.__name__}.{original_method.__name__}"
             ] = original_method
 
-            if inspect.iscoroutinefunction(original_method):
+            unwrapped_method = original_method
```

Comment:
We solved this in the previous version by unwrapping, but this check was failing for anthropic's async functions that were already wrapped